packages/neo-one-website/docs/0-installation/5-contract-templates.md

```diff
@@ -0,0 +1,106 @@
+---
+slug: contract-templates
+title: Contract Templates
+---
+
+NEO•ONE provides templates for common smart contract patterns.
+
+This guide will walk you through how to use a template.
+
+---
+
+[[toc]]
+
+---
+
+## Requirements
+
+Install the following packages if you have not done so:
+
+```bash
+yarn add @neo-one/client @neo-one/cli @neo-one/smart-contract @neo-one/smart-contract-test @neo-one/smart-contract-lib @neo-one/smart-contract-typescript-plugin
+```
+
+::: warning
+
+Important
+
+The package `@neo-one/smart-contract-lib` contains the templates and is important to this guide.
+
+:::
+
+---
+
+## Usage
+
+```typescript
+import { SmartContract } from '@neo-one/smart-contract';
+// import template of your choice from @neo-one/smart-contract-lib
+import { NEP5Token } from '@neo-one/smart-contract-lib';
+
+// Mixins
+export class YourContract extends NEP5Token(SmartContract) {
+  // 'SmartContract' here can be any other contract (abstract & non-abstract class) that extends SmartContract
+  // since NEP5Token returns an abstract class
+  // you need to implements the members specified as abstract in the returned NEP5TokenClass
+  // ... your smart contract code.
+}
+```
+
+If you find the syntax `NEP5Token(SmartContract)` strange, checkout this page on [Mixins](https://www.typescriptlang.org/docs/handbook/mixins.html) from the TypeScript Handbook for more details.
+
+Addtionally, you can also checkout this blog post - ["Real" Mixins with JavaScript Classes](https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/) - to see how mixins work under the hood.
+
+> A mixin is an abstract subclass; i.e. a subclass definition that may be applied to different superclasses to create a related family of modified classes.
+>
+> -Gilad Bracha and William Cook, [Mixin-based Inheritance](http://www.bracha.org/oopsla90.pdf)
+
+::: warning
+
+Tip
+
+It is encouraged that you spend time to explore how we implemented NEP5Token as well as other contract templates
+
+:::
+
+---
+
+## Example
+
+```typescript
+import { Address, Fixed, SmartContract } from '@neo-one/smart-contract';
+import { NEP5Token } from '@neo-one/smart-contract-lib';
+
+export class SimpleToken extends NEP5Token(SmartContract) {
+  public name: string = 'SimpleToken';
+  public symbol: string = 'STK';
+  public readonly owner: Address;
+  public readonly decimals: 8 = 8;
+
+  public constructor(owner: Address, amount: Fixed<8>) {
+    super();
+    if (!Address.isCaller(owner)) {
+      throw new Error('Sender was not the owner.');
+    }
+    this.owner = owner;
+    this.issue(owner, amount);
+  }
+}
+```
+
+---
+
+## Available Templates
+
+::: warning
+
+Note
+
+`NEP5Token` is being updated to `NEP17Token`. Stay tuned!
+
+:::
```

Comment:
I say we remove this to avoid confusion. The docs on the current website are for Neo2 only and NEP17 is for Neo3 only I believe.

If I'm wrong about that then disregard.